src/definition/guessfunc.go

```diff
@@ -0,0 +1,15 @@
+package definition
+
+import (
+	"fmt"
+	"strings"
+)
+
+// K8sMetricSetTypeGuesser is the metric set type guesser for k8s integrations.
+func K8sMetricSetTypeGuesser(_, groupLabel, _ string, _ RawGroups) (string, error) {
+	var sampleName string
+	for _, s := range strings.Split(groupLabel, "-") {
+		sampleName += strings.Title(s) //nolint: staticcheck // TODO: use golang.org/x/text/cases assuring no breaking changes.
```

Comment:
I don't think so. As I see it, it would require a specific task. It shouldn't be really complex (if both integration and e2e tests pass we should be fine) but I didn't want to do it as part of this PR to avoid making its scope even wider.